spec/models/section_spec.rb

```diff
@@ -142,6 +142,24 @@ def key_classes_for(hash)
     end
   end
 
+  describe '#save' do
+    it 'saves the last two editions' do
+      slug_generator = double(:slug_generator)
+      editions = [
+        edition_1 = double(:edition_1),
+        edition_2 = double(:edition_2),
+        edition_3 = double(:edition_3)
+      ]
+      section = Section.new(slug_generator, 'section-id', editions)
+
+      expect(edition_1).to_not receive(:save!)
+      expect(edition_2).to receive(:save!)
+      expect(edition_3).to receive(:save!)
```

Comment:
I rewrote the spec based on the current behaviour rather than the behaviour that was being implied by the previous spec. I would expect the spec the change again if/when we need to update the implementation.

I've added the following to the commit note to try to make this clearer:

> I've chosen to rewrite the tests rather than try to port them from section_repository_spec as I didn't think the existing ones were particularly good. The previous tests appear to imply some behaviour that doesn't actually exist in the implementation. The new tests describe the behaviour as it stands. The difference between the behaviour being described in the old tests and the actual behaviour might be indicative of a problem with the code but I'm happy to leave that for now.